community/detectors/apache_apisix_cve_2020_13945/src/main/java/com/google/tsunami/plugins/detectors/rce/cve202013945/Cve202013945Detector.java

```diff
@@ -0,0 +1,191 @@
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      targetUrlBuilder
+          .append("http://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    return targetUrlBuilder;
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2020-13945 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve202013945Detector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetVulnerabilityUrl = buildTarget(networkService).append(VUL_PATH).toString();
+    String randomVerifyPath = String.format("tsunami_%s", Instant.now(utcClock).toEpochMilli());
+    String targetExecuteUrl =
+        buildTarget(networkService)
+            .append(randomVerifyPath)
+            .append("?cmd=")
+            .append(URLEncoder.encode(EXECUTE_DATA, UTF_8))
+            .toString();
+    try {
+      HttpResponse httpResponse =
+          httpClient.sendAsIs(
+              post(targetVulnerabilityUrl)
+                  .setHeaders(
+                      HttpHeaders.builder()
+                          .addHeader(CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+                          .addHeader(TOKEN_HEADER_NAME, TOKEN_VALUE)
+                          .build())
+                  .setRequestBody(
+                      ByteString.copyFromUtf8(String.format(POST_DATA, randomVerifyPath)))
+                  .build());
+      if (httpResponse.status().code() == 201) {
+        logger.atInfo().log("Request payload to target %s succeeded", networkService);
+        HttpResponse executeResponse =
+            httpClient.sendAsIs(
+                get(targetExecuteUrl).setHeaders(HttpHeaders.builder().build()).build());
+        if (executeResponse.status().code() == 200
+            && executeResponse.bodyString().isPresent()
+            && executeResponse.bodyString().get().contains(DETECTION_STRING)) {
+          logger.atInfo().log("Vulnerability detected on target %s", networkService);
+          return true;
+        }
+      } else {
+        logger.atInfo().log(
+            "Execution of the command to the target %s has failed.", networkService);
+      }
+    } catch (IOException | AssertionError e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
```

Comment:
Do not log the entire `networkService` object, extract the target's basic info (such as hostname and port) instead